tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -143,24 +153,39 @@ ROOT::Experimental::RClusterDescriptor::RLocator
 ROOT::Experimental::Detail::RPageSinkDaos::CommitPageImpl(ColumnHandle_t columnHandle, const RPage &page)
 {
    auto element = columnHandle.fColumn->GetElement();
-   auto sealedPage = SealPage(page, *element, fOptions.GetCompression());
+   RPageStorage::RSealedPage sealedPage;
+   {
+      RNTupleAtomicTimer timer(fCounters->fTimeWallZip, fCounters->fTimeCpuZip);
+      sealedPage = SealPage(page, *element, fOptions.GetCompression());
+   }
 
    return CommitSealedPageImpl(columnHandle.fId, sealedPage);
 }
 
 
 ROOT::Experimental::RClusterDescriptor::RLocator
 ROOT::Experimental::Detail::RPageSinkDaos::CommitSealedPageImpl(
-   DescriptorId_t /*columnId*/, const RPageStorage::RSealedPage &sealedPage)
+   DescriptorId_t columnId, const RPageStorage::RSealedPage &sealedPage)
 {
-   auto offsetData = std::get<0>(fDaosContainer->WriteObject(sealedPage.fBuffer,
-                                                             sealedPage.fSize,
-                                                             kDistributionKey,
-                                                             kAttributeKey)).lo;
+   const auto bitsOnStorage = RColumnElementBase::GetBitsOnStorage(
+      fDescriptorBuilder.GetDescriptor().GetColumnDescriptor(columnId).GetModel().GetType());
+   const auto bytesPacked = (bitsOnStorage * sealedPage.fNElements + 7) / 8;
+
+   std::uint64_t offsetData;
+   {
+      RNTupleAtomicTimer timer(fCounters->fTimeWallWrite, fCounters->fTimeCpuWrite);
+      offsetData = std::get<0>(fDaosContainer->WriteObject(sealedPage.fBuffer,
+                                                           sealedPage.fSize,
+                                                           kDistributionKey,
+                                                           kAttributeKey)).lo;
+   }
 
    RClusterDescriptor::RLocator result;
    result.fPosition = offsetData;
    result.fBytesOnStorage = sealedPage.fSize;
+   fCounters->fNPageCommitted.Inc();
+   fCounters->fSzWritePayload.Add(sealedPage.fSize);
+   fCounters->fSzZip.Add(bytesPacked);
```

Comment:
I think `fSzSip` should include both packing and compression (like we do for the page source). Let's keep this counter in `CommitPageImpl`.

Reply:
Fixed, as discussed.